src/nft/utils/txRoute/combineItemsWithTxRoute.ts

```diff
@@ -36,14 +70,24 @@ export const combineBuyItemsWithTxRoute = (
     const newPriceInfo = item.updatedPriceInfo ? item.updatedPriceInfo : item.priceInfo
 
     // if the price changed
-    if (
-      route &&
-      'priceInfo' in route.assetOut &&
-      isPriceDiff(newPriceInfo.basePrice, route.assetOut.priceInfo.basePrice)
-    ) {
-      return {
-        ...item,
-        updatedPriceInfo: route.assetOut.priceInfo,
+    if (route && 'priceInfo' in route.assetOut) {
+      if (isPriceDiff(newPriceInfo.basePrice, route.assetOut.priceInfo.basePrice)) {
+        if (
+          !(
+            item.marketplace &&
+            isPooledMarket(item.marketplace) &&
+            isAveragePriceOfPooledAssets(
+              item,
+              items.filter((routeItem) => itemInRouteAndSamePool(item, routeItem, txRoute)).length,
+              route.assetOut.priceInfo.basePrice
```

Comment:
can we turn this into a callback/memo, something like `isAveragedPrice` to help clean up this conditional?